src/librustc_mir/interpret/memory.rs

```diff
@@ -423,10 +423,10 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> Memory<'a, 'mir, 'tcx, M> {
             if alloc.mutability == Mutability::Immutable {
                 return err!(ModifiedConstantMemory);
             }
-            let kind = M::STATIC_KIND.expect(
```

Comment:
cc @RalfJung

Reply:
So you are suing `M::STATIC_KIND = None` as indication for "statics may not be mutated"?

Seems reasonable, but the machine trait docs should reflect that explicitly.